app/models/manageiq/providers/azure/cloud_manager/provision/state_machine.rb

```diff
@@ -6,4 +6,21 @@ def customize_destination
 
     signal :post_create_destination
   end
+
+  def clone_failure_cleanup
+    delete_instance(phase_context[:clone_options])
```

Comment:
```suggestion
    delete_instance(phase_context[:clone_options]) if phase_context[:clone_options]
```

Reply:
I think we should signal_abort after this is successful, so maybe something like
```suggestion
    begin
      delete_instance(phase_context[:clone_options])
      signal_abort
    rescue
      requeue_phase(3.minutes)
    end
```

Reply:
> why do we need a requeue_phase here as well?

I think we should move the requeue_phase out of delete_instance and into here.  That way we keep all of the signal/requeue state directives in the same method